cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/LoadPersistedConcepts.kt

```diff
@@ -0,0 +1,318 @@
+            addEntriesFromFile(file, translationResult)
+        }
+    }
+
+    /**
+     * This function parses the provided [file] (containing the persisted [Concept]s) and adds the
+     * entries to the [translationResult].
+     *
+     * @param file The file containing the persisted concepts.
+     * @param translationResult The [TranslationResult] to which the [Concept]s will be added.
+     */
+    private fun addEntriesFromFile(file: File, translationResult: TranslationResult) {
+        val entries =
+            try {
+                val mapper =
+                    if (file.extension.lowercase() in listOf("yaml", "yml")) {
+                            ObjectMapper(YAMLFactory())
+                        } else {
+                            ObjectMapper(JsonFactory())
+                        }
+                        .registerKotlinModule()
+                mapper.readValue<PersistedConceptsEntry>(file)
+            } catch (ex: Exception) {
+                logger.error(
+                    "Error reading persisted concepts from ${file.path}: ${ex.message}",
+                    ex,
+                )
+                return
+            }
+
+        entries.concepts?.let { concepts ->
+            concepts.forEach { concept ->
+                if (concept.signature != null && concept.location != null) {
+                    log.error(
+                        "Both signature and location are set. Please use only one of them. The entire entry will be ignored!"
+                    )
+                    return@forEach
+                } else if (concept.signature != null) {
+                    getNodesBySignature(translationResult, concept.signature).forEach {
+                        underlyingNode ->
+                        addConcept(underlyingNode, concept.concept)
+                    }
+                } else if (concept.location != null) {
+                    getNodesByLocation(translationResult, concept.location).forEach { underlyingNode
+                        ->
+                        addConcept(underlyingNode, concept.concept)
+                    }
+                } else {
+                    log.error(
+                        "Neither signature nor location are set. The entire entry will be ignored!"
+                    )
+                }
+            }
+        }
+    }
+
+    /**
+     * This function retrieves the nodes matching the provided [signature] from the
+     * [translationResult]. Currently, this function only matches on the node being a
+     * [CallExpression] and the [CallExpression.reconstructedImportName] matching the provided FQN.
+     *
+     * @param translationResult The [TranslationResult] to search for nodes.
+     * @param signature The [SignatureEntry] containing the signature to match.
+     * @return A list of nodes matching the provided [SignatureEntry].
+     */
+    private fun getNodesBySignature(
```

Comment:
Could be made an extension function of `TranslationResult`
```suggestion
    private fun TranslationResult.getNodesBySignature(
```